internal/tasks/manifests.go

```diff
@@ -894,39 +900,84 @@ func (j *Janitor) CheckTrivySecurityStatusJob(registerer prometheus.Registerer)
 			},
 		},
 		BeginTx: j.db.Begin,
-		DiscoverRow: func(_ context.Context, tx *gorp.Transaction, _ prometheus.Labels) (securityInfo keppel.TrivySecurityInfo, err error) {
-			err = tx.SelectOne(&securityInfo, securityCheckSelectQuery, j.timeNow())
-			return securityInfo, err
+		DiscoverRow: func(_ context.Context, tx *gorp.Transaction, _ prometheus.Labels) (securityInfos []keppel.TrivySecurityInfo, err error) {
+			_, err = tx.Select(&securityInfos, securityCheckSelectQuery, j.timeNow())
+
+			// TODO: How to convince tx.Select to do this?
+			if len(securityInfos) == 0 {
+				err = sql.ErrNoRows
+			}
```

Comment:
@majewsky what am I doing wrong here?

Reply:
Nothing. `Select` (or its std counterpart `sql.Conn.Query`) can legitimately yield 0 rows (as opposed to `SelectOne` or `sql.Conn.QueryRow`), so that is not considered an error. The check as written here is fine.